src/platform/nm-linux-platform.c

```diff
@@ -1952,6 +2180,26 @@ _new_from_nl_link (NMPlatform *platform, const NMPCache *cache, struct nlmsghdr
 		}
 	}
 
+	if (obj->link.type == NM_LINK_TYPE_WIREGUARD) {
+		nm_auto_nmpobj NMPObject *lnk_data_now;
+
+		/* The WireGuard kernel module does not yet send link update
+		 * notifications, so we don't actually update the cache. For
+		 * now, always refetch link data here. */
+		lnk_data_now = nmp_object_new (NMP_OBJECT_TYPE_LNK_WIREGUARD, NULL);
+		if (!_wireguard_get_link_properties (platform, &obj->link, lnk_data_now)) {
```

Comment:
That came to mind, but didn't want to mess with `ext_data` after rebasing.

I'll go with whatever you guys decide here.

Reply:
when the branch was originally done, `ext_data` didn't even exist, so the question didn't come up.

I think, `NMPObjectLink.ext_data` and `NMPObjectLink.netlink.lnk` share some similarities, in that they are additional data that is attached to the `NMPObjectLink` data (there is also `NMPObjectLink.udev.device`). Optimally, the NMPObject would be only stupid data that we cache, usually obtained via netlink. In practice, that was never the case, because `NMPObjectLink.udev.device`. I think it makes actually sense, to attach all relevant bits to the `NMPObjectLink` objects, including the udev device, the ext-data, and the lnk-data. But while `NMPObjectLink` contains pointers to logic (udev-device, ext-data), the logic itself is not in `NMPObjectLink` itself, and that's the important part.

As it is, all fields in `NMP_OBJECT_TYPE_LNK_WIREGUARD` are plain data, and it thus fits well as it is. I don't think it's worth to convert NMPObjectLnkWireguard into an NMPExtDataWireguard, because the only logic associated with it, is `_wireguard_get_link_properties()`. `_wireguard_get_link_properties()` looks fine to me as it is.

Maybe the only thing that should indeed associated with the `NMPObjectLink` instance (alongside the ext-data), is the `wireguard_family_id`. By caching it in NMLinuxPlatformPrivate, it won't be correctly re-fetched, when the wireguard module is unloaded. By caching it inside NMPObjectLink, the lifetime of the `wireguard_family_id` is correctly limited to the lifetime of the interface (which cannot continue to exist after the module is unloaded).

Long story short, I think it is fine as is, but we should improve caching of `wireguard_family_id` in a future commit.

Reply:
> Long story short, I think it is fine as is, but we should improve caching of wireguard_family_id in a future commit.

OK.